Cargo.toml

```diff
@@ -143,6 +143,8 @@ debug = false
 [[bin]]
 name = "nu"
 path = "src/main.rs"
+bench = false
```

Comment:
Probably unrelated to the core of the PR. What would be the effect of doing this?

Reply:
Found this in [Criterion FAQ](https://bheisler.github.io/criterion.rs/book/faq.html#cargo-bench-gives-unrecognized-option-errors-for-valid-command-line-options).
It allows you to save and compare baselines when running benchmarks.  But it blocks Criterion from looking in your unit tests for `#[bench]`.  The only benchmarks run are the ones you put in separate `benches/` folder, and that's the way Nushell is currently structured.
Agree the change is not relevant to the current PR, I'll revert it.

Reply:
Moved that into a separate PR #8057 as it needs to affect all `Cargo.toml`s to take effect and we don't need to pollute the commit.